rotkehlchen/api/rest.py

```diff
@@ -1042,6 +1042,13 @@ def add_history_event(self, event: HistoryBaseEntry) -> Response:
 
     def edit_history_event(self, event: HistoryBaseEntry) -> Response:
         db = DBHistoryEvents(self.rotkehlchen.data.db)
+        # The extra_data field is something that we don't expose to the frontend in the case of
+        # evm transactions because it's a field that only we edit. In this case we need to be sure
+        # that when editing an event that this information is kept in the database and
+        # doesn't get overwritten.
+        # The identifier is not None here because it's sent by the frontend to edit the events and
+        # validated in the schema
+        event.extra_data = db.get_extra_data(event.identifier)  # type: ignore
```

Comment:
Weird how you thought of reading the DB, to take the already existing `extra_data`, add it to the event identifier, and rewrite it back to the DB.

Does that really sound smart to you? That's the only place that that `db.edit_history_event()`is called. At no point is an editing of extra data needed.

So why not just edit `db.edit_history_event()` to not touch extra data? And if we ever need one to do that, we make another function then?

![giphy (10)](https://user-images.githubusercontent.com/1658405/223876135-daf42708-3832-45d9-b552-07e7d41cb82a.gif)